src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -569,6 +581,74 @@ public FileInfoFormatting()
             }
         }
 
+        /// <summary>
+        /// Contains formatting styles for user prompts.
+        /// </summary>
+        public sealed class PromptFormatting
+        {
+            // TODO: use PSStyle.Foreground
+            // TODO: create a class AnsiEscape with ValidateNoContent in string ctor
```

Comment:
This is something I would like to do in separate MR, as it should remove a lot of boilerplate from this code and `PowerShellCore_format_ps1xml.cs`

Reply:
Sure, please make sure to create a new issue for tracking and then we can go from there

Reply:
Will do after merge, will wait because if some reviewer finds a good reason not to do it, then there is no point keeping the TODOs

Reply:
What is the intent of the `AnsiEscape` class you have TODO that StringDecorated doesn't provide?

Reply:
`StringDecorated` has plaintext + escapes, `AnsiEscape` (name bikeshed) contains only escape. I want it to integrate the `CheckNoPlaintext` in its string constructor and add `ToVisibleString` equivalent to current display method. That way we don't need to repeat the same thing for every variable in `PSStyle`.